brave-instrumentation/aws-java-sdk-core/src/main/java/brave/instrumentation/aws/TracingRequestHandler.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright 2016-2018 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.instrumentation.aws;
+
+import brave.Span;
+import brave.Tracing;
+import brave.http.HttpClientAdapter;
+import brave.http.HttpClientHandler;
+import brave.http.HttpTracing;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.AmazonWebServiceResult;
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import com.amazonaws.ResponseMetadata;
+import com.amazonaws.handlers.HandlerAfterAttemptContext;
+import com.amazonaws.handlers.HandlerContextKey;
+import com.amazonaws.handlers.RequestHandler2;
+import zipkin2.Endpoint;
+
+/**
+ * Traces AWS Java SDK calls. Adds on the standard zipkin/brave http tags, as well as tags that
+ * align with the XRay data model.
+ */
+public final class TracingRequestHandler extends RequestHandler2 {
+
+  public static TracingRequestHandler create(HttpTracing httpTracing) {
+    return new TracingRequestHandler(httpTracing);
+  }
+
+  public static TracingRequestHandler create(Tracing tracing) {
+    return new TracingRequestHandler(HttpTracing.create(tracing));
+  }
+
+  static final HandlerContextKey<Span> SPAN = new HandlerContextKey<>(Span.class.getCanonicalName());
+  static final HandlerContextKey<TracingRequestHandler> TRACING_REQUEST_HANDLER_CONTEXT_KEY = new HandlerContextKey<>(TracingRequestHandler.class.getCanonicalName());
+
+  static final HttpClientAdapter<Request<?>, Response<?>> ADAPTER = new HttpAdapter();
+
+  static final Propagation.Setter<Request<?>, String> SETTER = new Propagation.Setter<Request<?>, String>() {
+    @Override public void put(Request<?> carrier, String key, String value) {
+      carrier.addHeader(key, value);
+    }
+  };
+
+  final HttpClientHandler<Request<?>, Response<?>> handler;
+  final TraceContext.Injector<Request<?>> injector;
+
+  TracingRequestHandler(HttpTracing httpTracing) {
+    handler = HttpClientHandler.create(httpTracing, ADAPTER);
+    injector = httpTracing.tracing().propagation().injector(SETTER);
+  }
+
+  @Override public final void beforeRequest(Request<?> request) {
+    Span span = handler.handleSend(injector, request);
+    span.remoteEndpoint(
+        Endpoint.newBuilder().serviceName(request.getServiceName()).build());
+    span.tag("aws.service_name", request.getServiceName());
```

Comment:
I wonder if we should create a separate `HttpClientParser` to handle this stuff. I think because this is RPC over HTTP we should probably just make the Handler own this logic, and allow extending, for things like Dynamo and SQS with ClientParsers

Reply:
> yep was thinking similar.. especially if http under ends up instrumented

I think I need you to clarify which part you are talking about